django_comparison_dashboard/graphs.py

```diff
@@ -191,6 +192,21 @@ def sankey(data, filter_set: SankeyGraphFilterSet):
 
         value.append(flow["value"])
 
+    node_colors = [
+        f'rgba({int(color_dict[label][1:3], 16)}, {int(color_dict[label][3:5], 16)}, {int(color_dict[label][5:7], 16)}, 0.75)'
+        if label in color_dict else
+        f'rgba({random.randint(0, 255)}, {random.randint(0, 255)}, {random.randint(0, 255)}, 0.75)'
+        for label in labels
+    ]
+
+    # Map colors to links based on their source node with reduced opacity (alpha = 0.5)
+    link_colors = [
```

Comment:
I also thought, that we should color links in same color as the source.
But this could be done without looking up color_dict or random, by storing sources and related colors beforehand and set color of links according to source, right? 
Then, colors are adapted for links, even if color cannot be found in color_dict.

Reply:
The advantage would be, that also processes and not only commodities that are defined in the dict would have the same outgoing colors, as I understand?

I would then still use a dictionary to define fixed colors based on the commodity-colors that we defined in the "Modellstruktur"- but this shouldn't be a problem, right?